compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -1321,7 +1328,9 @@ impl<'v> RootCollector<'_, 'v> {
 
         let start_def_id = match self.tcx.lang_items().require(LangItem::Start) {
             Ok(s) => s,
-            Err(err) => self.tcx.sess.fatal(&err),
+            Err(error_message) => {
+                self.tcx.sess.emit_fatal(FatalError { error_message: error_message.clone() });
+            }
         };
```

Comment:
For now, just change `require` to return some appropriate error type which isn't the English error message, e.g. `struct LangItemError(LangItem)`. Add a function to that new type that returns the current `String` - use that everywhere else to avoid having to make changes to lots of other use sites. Implement `IntoDiagnosticArg` for it so that you can just use it in the `SessionDiagnostic` in this crate.

Reply:
Excellent idea!

✅ Done